be/src/exec/pipeline/pipeline_driver.h

```diff
@@ -274,7 +274,21 @@ class PipelineDriver {
 
     // return true if either dependencies_block or local_rf_block return true, which means that the current driver
     // should wait for both hash table and local runtime filters' readiness.
-    bool is_precondition_block() { return dependencies_block() || local_rf_block() || global_rf_block(); }
+    bool is_precondition_block() {
+        if (!_wait_global_rf_ready) {
+            if (dependencies_block() || local_rf_block()) {
+                return true;
+            }
+            _wait_global_rf_ready = true;
+            if (_global_rf_descriptors.empty()) {
+                return false;
+            }
+            _global_rf_wait_timeout_ns += _precondition_block_timer_sw->elapsed_time() + 5000000;
```

Comment:
Why not directly change `_global_rf_wait_timeout_ns`?

Reply:
according to RuntimeFilterProbeCollector::wait(), actually, it wait for 25ms. so 5ms is added